src/daft-connect/src/translation/logical_plan/join.rs

```diff
@@ -0,0 +1,76 @@
+use daft_logical_plan::LogicalPlanBuilder;
+use eyre::{bail, WrapErr};
+use spark_connect::join::JoinType;
+use tracing::warn;
+
+use crate::translation::{to_daft_expr, to_logical_plan};
+
+pub fn join(join: spark_connect::Join) -> eyre::Result<LogicalPlanBuilder> {
+    let spark_connect::Join {
+        left,
+        right,
+        join_condition,
+        join_type,
+        using_columns,
+        join_data_type,
+    } = join;
+
+    let Some(left) = left else {
+        bail!("Left side of join is required");
+    };
+
+    let Some(right) = right else {
+        bail!("Right side of join is required");
+    };
+
+    if let Some(join_condition) = join_condition {
```

Comment:
so we'll need some complicated logic to split the join condition into left and right halves. Check out the SQL frontend. We're doing something similar to split the join condition